test/vmtests/vmtests/utils/libvirt_utils.py

```diff
@@ -136,21 +155,57 @@ def _add_disk_xml(
     disk_source = ET.SubElement(disk, "source")
     disk_source.attrib["file"] = file_path
 
+# Adds a disk to a libvirt domain XML document.
+def _add_iso_xml(
+    devices: ET.Element,
+    file_path: str,
+    bus_type: str,
+    next_disk_indexes: Dict[str, int],
+) -> None:
+    device_name = "hda"
+
+    disk = ET.SubElement(devices, "disk")
+    disk.attrib["type"] = "file"
+    disk.attrib["device"] = "cdrom"
+
+    disk_driver = ET.SubElement(disk, "driver")
+    disk_driver.attrib["name"] = "qemu"
+    disk_driver.attrib["type"] = "raw"
+
+    disk_target = ET.SubElement(disk, "target")
+    disk_target.attrib["dev"] = device_name
+    disk_target.attrib["bus"] = bus_type
+
+    disk_source = ET.SubElement(disk, "source")
+    disk_source.attrib["file"] = file_path
+    disk_source.attrib["index"] = "1"
+
+    disk_readonly = ET.SubElement(disk, "readonly")
+
+    disk_alias = ET.SubElement(disk, "alias")
+    disk_alias.attrib["name"] = "ide0-0-0"
```

Comment:
Make sure you shut down a libvirt VM before copying its XML.

These aren't needed:

- `source['index']`
- `alias`

Reply:
pending

Reply:
Which part are you referring to re. 'copying its XML'?

Reply:
I am guessing that to see how to provision a CDROM device, you created a VM using virt-manager or virt-install and then looked at the VM's xml. However, if the VM is running, the XML will contain runtime metadata/status which isn't needed to create the VM.